pinot-core/src/main/java/org/apache/pinot/core/data/readers/PinotSegmentRecordReader.java

```diff
@@ -110,7 +136,7 @@ public PinotSegmentRecordReader(File indexDir, @Nullable Schema schema, @Nullabl
         _docIdsInSortedColumnOrder = null;
       }
     } catch (Exception e) {
-      _immutableSegment.destroy();
+      _indexSegment.destroy();
```

Comment:
We don't want to destroy the segment if it is passed in instead of loaded in this class